src/parser/bison_parser.y

```diff
@@ -1012,6 +1023,39 @@ opt_alias:
 	|	/* empty */ { $$ = nullptr; }
 
 
+/******************************
+ * With Descriptions
+ ******************************/
+
+opt_with_clause:
+		with_clause
+	| 	/* empty */ { $$ = nullptr; }
+	;
+
+with_clause:
+		WITH with_description_list { $$ = $2; }
+	;
+
+with_description_list:
+		with_description {
+			$$ = new std::vector<WithDescription*>();
+			$$->push_back($1);
+		}
+	|	with_description_list ',' with_description {
+			$1->push_back($3);
+                        $$ = $1;
+		}
+	;
+
+with_description:
+		IDENTIFIER AS select_with_paren {
+			$$ = new WithDescription();
+			$$->alias = $1;
```

Comment:
Are you sure that we can claim ownership over that string? Other places in here use `strdup`.

Reply:
I think we can. In our rule/action-definitions, `strdup` is always used in combination with the `delete` command, which we do not use in this PR. For example:
```bison
file_path:
		string_literal { $$ = strdup($1->name); delete $1; }
	;
string_literal:
		STRING { $$ = Expr::makeLiteral($1); }
	;
```
I also did not find any warnings regarding this in the O`Reilly Flex/Bison book. At least not for rule actions.
Correct me if I'm wrong.

Reply:
Then who owns the string and makes sure that it exists for as long as the SelectStatement does?

Reply:
I think it is our job to manage that, e.g.:
```cpp
  WithDescription::~WithDescription() {
    free(alias);
    delete select;
  }
```

Reply:
got it